lib/metrics.rb

```diff
@@ -0,0 +1,183 @@
+# frozen_string_literal: true
+
+module Metrics
+  class TimeSeries
+    attr_accessor :labels, :query
+
+    def initialize(labels:, query:)
+      @labels = labels
+      @query = query
+    end
+  end
+
+  class MetricDefinition
+    attr_accessor :name, :description, :unit, :series
+
+    def initialize(name:, description:, unit:, series:)
+      @name = name
+      @description = description
+      @unit = unit
+      @series = series
+    end
+  end
+
+  POSTGRES_METRICS = [
```

Comment:
We often map or find by name. Should we use a hash instead of an array, like `{ name: details }`? This way, we can get the list of metric names with `POSTGRES_METRICS.keys` or find a specific metric with `POSTGRES_METRICS[name]`